community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -753,17 +759,31 @@ public void close( Consumer<PageCursor> headerWriter ) throws IOException
 
     private void close( Header.Writer headerWriter ) throws IOException
     {
-        writer.close();
-
+        writerCheckpointMutex.lock();
```

Comment:
So `writer.close()` was not protected from concurrent writes before?
Lets add a test for this!